### PR TITLE
Make liveness and readiness settings configurable for the metrics con…

### PR DESCRIPTION
**Pull request by svein-hallsteinsen**
Need to make liveness / readiness settings configurable for the metrics container.
Background: Low timeout values may mark the metrics container unavailable, which in turn means new connections to mysql cannot be made.

### CODE REVIEW

**Comment by whereisaaron** [comment]
Thanks for this PR @svein-hallsteinsen I was coming here specifically to propose the same fix!

The default 1s readiness probe time for the metrics sidecar seems to low in practice. Maybe if the probe was ':9104/' rather than ':9104/metrics' the response would be fast enough. Otherwise 2s might be better. Either way, we should be able to adjust it with the chart.

**Comment by svein-hallsteinsen** [comment]
Hi @whereisaaron , in case you (and others) need it this is the workaround we currently use:

kubectl patch deployment -n mysql mysql -p '{"spec": {"template": {"spec": {"containers": [{"name": "metrics","readinessProbe": {"timeoutSeconds": 5},"livenessProbe": {"timeoutSeconds": 5}}]}}}}'

Add this patch after installing the chart. This will set timeouts to 5 seconds, running the patch will restart mysql if values change. We have the same issue as you, a readiness timeout of 1 seconds sometimes is too little. Sometimes it's enough and other times it's not.

Hopefully someone will look at this PR so we can avoid ugly hacks :-)

**Comment by whereisaaron** [comment]
Thanks for the workaround patch @svein-hallsteinsen hopefully this small fix/patch will get merged pretty quickly since it causes operational errors (unless you disable metrics entirely).

File: stable/mysql/templates/deployment.yaml [hunk 1]
```diff
@@ -133,14 +133,14 @@ spec:
           httpGet:
             path: /metrics
             port: metrics
-          initialDelaySeconds: 15
-          timeoutSeconds: 5
+          initialDelaySeconds: {{ .Values.metrics.livenessProbe.initialDelaySeconds }}
```
Comment (paulczar) [inline stable/mysql/templates/deployment.yaml:140]: please add this variable to README.md

File: stable/mysql/templates/deployment.yaml [hunk 2]
```diff
@@ -133,14 +133,14 @@ spec:
...
+          timeoutSeconds: {{ .Values.metrics.livenessProbe.timeoutSeconds }}
```
Comment (paulczar) [inline stable/mysql/templates/deployment.yaml:141]: please add this variable to README.md

File: stable/mysql/templates/deployment.yaml [hunk 3]
```diff
@@ -133,14 +133,14 @@ spec:
...
         readinessProbe:
           httpGet:
             path: /metrics
             port: metrics
-          initialDelaySeconds: 5
-          timeoutSeconds: 1
+          initialDelaySeconds: {{ .Values.metrics.readinessProbe.initialDelaySeconds }}
```
Comment (paulczar) [inline stable/mysql/templates/deployment.yaml:146]: please add this variable to README.md

File: stable/mysql/templates/deployment.yaml [hunk 4]
```diff
@@ -133,14 +133,14 @@ spec:
...
+          timeoutSeconds: {{ .Values.metrics.readinessProbe.timeoutSeconds }}
```
Comment (paulczar) [inline stable/mysql/templates/deployment.yaml:147]: please add this variable to README.md

**Review by paulczar — CHANGES_REQUESTED** [review]
Please bump the chart version (probably should be a minor bump) and update the README to add the new values to the table.

**Comment by svein-hallsteinsen** [comment]
Thanks for your feedback @paulczar - I have pushed the changes you requested now. I'm new to this so I hope I'm doing it the right way.

**Comment by svein-hallsteinsen** [comment]
Hi @scottrigby - any chance you can take a look at this? Very small pull request.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**